CONTRIBUTING.md

```diff
@@ -0,0 +1,180 @@
+
+   ```shell
+   git switch -c my_branch  # Pick a name appropriate to your work
+   git pull upstream main
+   ```
+
+2. Make changes and commit to local branch.
+
+   ```shell
+   # ... editing, testing, ... 
+   git commit ...
+   ```
+
+3. Pull any changes that may have been made in the upstream repository
+   main branch.
+
+   ```shell
+   git switch my_branch
+   git pull --rebase upstream main
+   ```
+
+   Note that this command may result in merge conflicts. Fix those if needed.
+
+4. Push your branch to the corresponding branch in your fork (the `origin` repository).
+
+   ```shell
+   git switch my_branch
+   git push origin my_branch
+   ```
+
+5. Select the branch you are working on in the drop-down menu of branches on
+   https://github.com/_your_user_name_/weaver-kube . Then hit the `Compare and pull
+   request` button.
+
+6. Respond to feedback, which may involve making new commits.
+   If you made any changes, push them to github again.
+
+   ```shell
+   git switch my_branch
+   git push origin my_branch
+   ```
+
+   Repeat as necessary until all feedback has been handled.
+
+   Note: the preceding approach will cause the pull request to become a sequence
+   of commits. Some people like to keep just a single commit that is amended as
+   changes are made. If you are amending commits that had already been pushed,
+   you will have to add `--force` to the `git push` command above.
+
+7. Once reviewers are happy, pull any main branch changes that may
+   have happened since step 3.
+
+    ```shell
+    git switch my_branch
+    git pull --rebase upstream main
+    ```
+
+   If some changes were pulled, push again to the PR, but this time you will
+   need to force push since the rebase above will have rewritten your commits.
+
+    ```shell
+    git switch my_branch
+    git push --force origin my_branch
+    ```
+
+8.  Ask somebody who has permissions (or do it yourself if you
+    have permissions) to merge your branch into the main branch
+    of the `upstream` repository. The reviewer may do this without
+    being asked.
+
+    Select the `Squash and merge` option on https://github.com/ServiceWeaver/weaver-kube
+    or use the command line instructions found on that page. Edit the commit message
+    as appropriate for the squashed commit.
+
+9.  Delete the branch from `origin`:
+
+    ```
+    git push origin --delete my_branch
+    ```
+
+10. Delete the branch from `local`
+
+    ```
+    git switch main
+    git branch -D my_branch
+    ```
+
+### Code Reviews
+
+All submissions, including submissions by project members, require review. We
+use GitHub pull requests for this purpose. Consult [GitHub Help][github_help]
+for more information on using pull requests.
+
+To make sure changes are well coordinated, we ask you to discuss any significant
+change prior to sending a pull request. To do so, either file a
+[new issue][new_issue] or claim an [existing one][issues].
+
+## What to contribute to?
```

Comment:
I think this `What to contribute to` section is a bit stale? Maybe delete it or update it to be weaver-kube specific?

Reply:
Done. Copied blindly from /weaver.